command/secrets/output/params.go

```diff
@@ -0,0 +1,248 @@
+	}
+
+	op.secretsManager = secretsManager
+
+	return nil
+}
+
+func (op *outputParams) parseConfig() error {
+	secretsConfig, readErr := secrets.ReadConfig(op.configPath)
+	if readErr != nil {
+		return errInvalidConfig
+	}
+
+	if !secrets.SupportedServiceManager(secretsConfig.Type) {
+		return errUnsupportedType
+	}
+
+	op.secretsConfig = secretsConfig
+
+	return nil
+}
+
+func (op *outputParams) initLocalSecretsManager() error {
+	validatorPathPrefix := filepath.Join(op.dataDir, secrets.ConsensusFolderLocal)
+	networkPathPrefix := filepath.Join(op.dataDir, secrets.NetworkFolderLocal)
+	dataDirAbs, _ := filepath.Abs(op.dataDir)
+
+	if _, err := os.Stat(op.dataDir); os.IsNotExist(err) {
+		return fmt.Errorf("the data directory provided does not exist: %s", dataDirAbs)
+	}
+
+	errs := make([]string, 0, 2)
+	if _, err := os.Stat(validatorPathPrefix); os.IsNotExist(err) {
+		errs = append(errs, fmt.Sprintf("no validator keys found in the data directory provided: %s", dataDirAbs))
+	}
+
+	if _, err := os.Stat(networkPathPrefix); os.IsNotExist(err) {
+		errs = append(errs, fmt.Sprintf("no network key found in the data directory provided: %s", dataDirAbs))
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf(strings.Join(errs, "\n"))
+	}
+
+	local, err := helper.SetupLocalSecretsManager(op.dataDir)
+	if err != nil {
+		return err
+	}
+
+	op.secretsManager = local
+
+	return nil
+}
+
+func (op *outputParams) initValidatorAddress() error {
+	validatorAddress, err := helper.LoadValidatorAddress(op.secretsManager)
+	if err != nil {
+		return err
+	}
+
+	if validatorAddress == types.ZeroAddress {
+		op.validatorAddress = ""
+	} else {
+		op.validatorAddress = validatorAddress.String()
+	}
+
+	return nil
+}
+
+func (op *outputParams) initBLSPublicKey() error {
+	blsPubkey, err := helper.LoadBLSPublicKey(op.secretsManager)
+	if err != nil {
+		return err
+	}
+
+	op.blsPubkey = blsPubkey
+
+	return nil
+}
+
+func (op *outputParams) initNodeID() error {
+	nodeID, err := helper.LoadNodeID(op.secretsManager)
+	if err != nil {
+		return err
+	}
+
+	op.nodeID = nodeID
+
+	return nil
+}
+
+func (op *outputParams) getResult() command.CommandResult {
+	if op.outputNodeID {
+		return &SecretsOutputResult{
+			NodeID: op.nodeID,
+
+			outputValidator: op.outputValidator,
+			outputBLS:       op.outputBLS,
+			outputNodeID:    op.outputNodeID,
+		}
+	}
+
+	if op.outputValidator {
+		return &SecretsOutputResult{
+			Address: op.validatorAddress,
+
+			outputValidator: op.outputValidator,
+			outputBLS:       op.outputBLS,
+			outputNodeID:    op.outputNodeID,
+		}
+	}
+
+	if op.outputBLS {
+		return &SecretsOutputResult{
+			BLSPubkey: op.blsPubkey,
+
+			outputValidator: op.outputValidator,
+			outputBLS:       op.outputBLS,
+			outputNodeID:    op.outputNodeID,
+		}
+	}
+
+	return &SecretsOutputResult{
+		Address:   op.validatorAddress,
+		BLSPubkey: op.blsPubkey,
+		NodeID:    op.nodeID,
+
+		outputValidator: op.outputValidator,
+		outputNodeID:    op.outputNodeID,
+	}
+}
```

Comment:
There are many conditional branches in this function and `GetOutput`. You might want to define 2 result structs: one is the struct that has only one field and another is the struct that has all public keys and ID.

Reply:
I compacted the conditionals a bit to fix my laziness. Thanks for the suggestion. Let me know if you have any better direction 🙏